packages/elements/src/tree/elements/tree-item.ts

```diff
@@ -156,24 +157,69 @@ export class TreeItem<T extends TreeDataItem = TreeDataItem> extends ControlElem
     return this.checkedState === CheckedState.INDETERMINATE;
   }
 
+  /**
+   * Handles aria-checked and aria-selected on mode changes
+   * aria-checked is used for multiple mode due to tri-state support
+   * @returns {void}
+   **/
+  private multipleChanged (): void {
+    this.removeAttribute(this.multiple ? 'aria-selected' : 'aria-checked');
+    this.checkedChanged();
+  }
+
+  /**
+   * Handles selectedd and arias attribute changes
+   * @returns {void}
+   */
+  private checkedChanged (): void {
+    switch (this.checkedState) {
+      case CheckedState.CHECKED:
+        this.setAttribute('selected', '');
+        this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'true');
+        break;
+      case CheckedState.INDETERMINATE:
+        this.setAttribute('selected', 'indeterminate');
+        this.setAttribute('aria-checked', 'mixed');
+        break;
+      default:
+        this.removeAttribute('selected');
+
+        if (this.multiple) {
+          this.setAttribute('aria-checked', 'false');
+        }
+
+        // In single mode, only children nodes are selectable
+        if (!this.parent && !this.multiple) {
+          this.setAttribute('aria-selected', 'false');
+        }
+        break;
+    }
+  }
```

Comment:
@wsuwt I feel this should always be linked with the selected state, so if `selected` attribute is being removed, the `aria-selected` should be set to `false`. The logic for setting the correct checked state can be handled in the renderer.

```suggestion
  private checkedChanged (): void {
    switch (this.checkedState) {
      case CheckedState.CHECKED:
        this.setAttribute('selected', '');
        this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'true');
        break;
      case CheckedState.INDETERMINATE:
        this.setAttribute('selected', 'indeterminate');
        this.setAttribute('aria-checked', 'mixed');
        break;
      default:
        this.removeAttribute('selected');
        this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'false');
        break;
    }
  }
```

Reply:
Fixed